filebeat/input/log/input.go

```diff
@@ -224,6 +228,8 @@ func (p *Input) loadStates(states []file.State) error {
 
 // Run runs the input
 func (p *Input) Run() {
+	p.healthy = true
+	p.updateStatus(status.Starting, "starting the scan")
```

Comment:
I'd say that by the time `Run` starts the input is already running. Also `Run` is called regurlary while the input is running, so that will cause it to transition from Running to Starting again.

I suggest putting the status Starting at the beginning of `NewInput`, one know issue with the Log input (that is solved by Filestream) is that if registry is large, reading the states from the registry at the input start up can be slow. I believe this is a good use of the Starting state. The sates are loaded here: https://github.com/elastic/beats/blob/99b5625e149797574611d5181f0985f410319192/filebeat/input/log/input.go#L173

Reply:
> I suggest putting the status Starting at the beginning of NewInput

Makes sense!